storage/badger/operation/common.go

```diff
@@ -115,6 +115,26 @@ func update(key []byte, entity interface{}) func(*badger.Txn) error {
 	}
 }
 
+// upsert will encode the given entity with MsgPack and upsert the binary data
+// under the given key in the badger DB.
+func upsert(key []byte, entity interface{}) func(*badger.Txn) error {
+	return func(tx *badger.Txn) error {
+		// serialize the entity data
```

Comment:
`insert` code has some cargo-culting around ``len(key)` and `SetMax`.  Do we want to keep it?

Reply:
good catch - yes we do need it when the key being upserted is new